### DISCUSSION TITLE
Multiple automerges

### DISCUSSION THREAD

**Discussion by bpfoster** [renovatebot/renovate#9670] [Request Help]
If I read https://docs.renovatebot.com/automerge-configuration/ , it sounds to me like if there are multiple automerge candidates, renovate should do one, then rebase the others, then move on to the next, etc, all on the same run until done.

I'm seeing a different behavior (self-hosted, docker), where it will automerge the first one, then stop doing any more branches. 

The next run of renovate will do the next automerge, and stop.  Rinse and repeat until they're finally all done.  For a large amount of candidate updates, this therefore takes a long time to work through.

Is this the expected behavior, do I have something misconfigured?

**Comment by viceice** [comment]
This is the expected behavior

**Reply by rarkins** [reply]
FYI the "automatically" there is about the app, which uses web hooks to detect the first automerge and then queues a second job to run. It doesn't apply to self-hosted

**Reply by bpfoster** [reply]
Ok, thanks for the clarification.

**Reply by pqvst** [reply]
Is there any way to make a renovate (self-hosted as a github action) automerge multiple branches in one go? Or auto queue up another job if there are more pending automerges?

**Reply by viceice** [reply]
nope, renovate will ever only automerge one branch at a run, but it does a second rund directly after to rebase other pr's. because renovate assumes rebase branches are pending for ci checks it never tries to automerge after rebase. So you always need another run for next auftomerge.

**Reply by rarkins** [reply]
Are you skipping all tests? 

Otherwise, how would you imagine it could work when Renovate needs to wait for tests to pass for the rebased branches?
